Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -2249,6 +2252,30 @@ static void webkit_web_view_class_init(WebKitWebViewClass* webViewClass)
         g_cclosure_marshal_generic,
         G_TYPE_BOOLEAN, 1,
         WEBKIT_TYPE_USER_MESSAGE);
+
+    /**
+     * WebKitWebView::query-permission:
+     * @web_view: the #WebKitWebView on which the signal is emitted
+     * @request: the #WebKitQueryPermissionRequest
+     *
+     * This signal allows the User-Agent to respond to permission requests for powerful features, as
+     * specified by the [Permissions W3C Specification](https://w3c.github.io/permissions/).
+     *
+     * If the signal is not handled, the query result will be set to %WEBKIT_QUERY_PERMISSION_PROMPT automatically.
+     *
```

Comment:
Explain here that this can be handled asynchronously by raking a ref of the query and returning `FALSE`. Check other signals.